[rpc] Remove describeLakeStorage to get LakeStorageInfo from table properties

SteNicholas opened the pull request:
### Purpose

Linked issue: close #411

`LakeStorageInfo` is forwarded info to table properties in #362. Therefore, `describeLakeStorage` could be removed since `LakeStorageInfo` has been in table properties.

### Brief change log

1. Remove `describeLakeStorage` to get `LakeStorageInfo` from table properties via `LakeStorageUtils#getLakeStorageInfo`.
2. Remove `lakeStorageInfo` in `LakeSnapshot`.
3. Fix `MetadataManager#getTable(TablePath tablePath)` to fill in `TableInfo` with datalake properties.

### Tests

CI.

### API and Format

Remove `Admin#describeLakeStorage`.

### Documentation

No.

SteNicholas commented:
Ping @wuchong, @luoyuxia.

luoyuxia commented on fluss-connectors/fluss-connector-flink/src/main/java/com/alibaba/fluss/connector/flink/catalog/FlinkCatalog.java:
```diff
@@ -530,13 +532,14 @@ protected TablePath toTablePath(ObjectPath objectPath) {
 
-    private void mayInitLakeCatalogCatalog() {
+    private void mayInitLakeCatalogCatalog(TableInfo tableInfo) {
         if (lakeCatalog == null) {
```
nit: 
Currently, the logic of initializing `lakeCatalog`  rely all tables should have same datalake confgiuration.. Although currently it's the truth. Do we need add some comment in here in case it breaks in some day?

luoyuxia reviewed (commented):
@SteNicholas Thanks for the pr. Left some comments..
Also, since the lake storage info can be retrieved from table, the `LakeStorageInfo` in `LakeSnapshot` can also be removed.

SteNicholas commented:
@wuchong, @luoyuxia, I have addressed above comments in https://github.com/alibaba/fluss/pull/503/commits/f14578e17edf265c94f8fe8bf536a3679e4c5b18. PTAL.

wuchong reviewed (approved):
I added following changes to the pull request:

1. The `PbLakeStorageInfo` has been removed as it is no longer used.
2. The `MetadataManager` now requires a mandatory cluster config parameter to avoid missing inputs in places where it is used. (Previously, the `getTable` method had a parameterless version, which led to some places omitting the parameter.)
3. For tables where the data lake feature is not enabled, the lake catalog properties will not be automatically filled in.
